packages/kbn-es/src/cluster.ts

```diff
@@ -575,35 +586,40 @@ exports.Cluster = class Cluster {
 
   /**
    * Run an Elasticsearch Serverless Docker cluster
-   *
-   * @param {ServerlessOptions} options
+   * @param options ServerlessOptions
+   * @returns node names
    */
-  async runServerless(options = {}) {
-    if (this._process || this._outcome) {
-      throw new Error('ES has already been started');
+  async runServerless(options: ServerlessOptions) {
+    if (this.process || this.outcome) {
+      throw new Error('ES stateful cluster has already been started');
     }
 
-    this._serverlessNodes = await runServerlessCluster(this._log, options);
+    if (this.serverlessNodes.length > 0) {
+      throw new Error('ES serverless docker cluster has already been started');
+    }
+
+    this.serverlessNodes = await runServerlessCluster(this.log, options);
 
     if (options.teardown) {
       /**
        * Ideally would be async and an event like beforeExit or SIGINT,
        * but those events are not being triggered in FTR child process.
        */
-      process.on('exit', () => teardownServerlessClusterSync(this._log, options));
+      process.on('exit', () => teardownServerlessClusterSync(this.log, options));
```

Comment:
I tried to cover it with integration test, but didn't find a working solution. Open to discuss it